File: src/main/java/com/polygloat/exceptions/InternalException.java
package com.polygloat.exceptions;

import com.polygloat.constants.Message;
import org.springframework.http.HttpStatus;

import java.io.Serializable;
import java.util.List;

public class InternalException extends ErrorException {
    public InternalException(Message message, List<Serializable> params) {
        super(message, params);
    }

    public InternalException(Message message) {
        super(message);
    }

    public HttpStatus getHttpStatus() {
        return HttpStatus.INTERNAL_SERVER_ERROR;
    }
}

File: src/main/kotlin/io/tolgee/exceptions/InternalException.kt
package io.tolgee.exceptions

import io.tolgee.constants.Message
import org.springframework.http.HttpStatus
import java.io.Serializable

class InternalException : ErrorException {
  constructor(message: Message, params: List<Serializable>?) : super(message, params) {}
  constructor(message: Message) : super(message) {}

  override val httpStatus: HttpStatus?
    get() = HttpStatus.INTERNAL_SERVER_ERROR
}
